version: '3.8'

services:
  postgres:
    image: postgres:13
    container_name: postgres
    env_file: .env
    volumes:
      - postgres-db-volume:/var/lib/postgresql/data
      - ./docker/docker-entrypoint-initdb.d:/docker-entrypoint-initdb.d
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "${POSTGRES_USER}"]
      interval: 10s
      retries: 5
      start_period: 5s
    restart: always
    networks:
      - my_network

  redis:
    image: redis:7
    container_name: redis
    env_file: .env
    expose:
      - 6379
    healthcheck:
      test: ["CMD", "redis-cli", "ping"]
      interval: 10s
      timeout: 30s
      retries: 50
      start_period: 30s
    restart: always
    volumes:
      - redis:/data
    networks:
      - my_network

  pgadmin:
    container_name: pgadmin
    image: dpage/pgadmin4:7.6
    restart: always
    environment:
      - PGADMIN_DEFAULT_EMAIL=${PGADMIN_DEFAULT_EMAIL}
      - PGADMIN_DEFAULT_PASSWORD=${PGADMIN_DEFAULT_PASSWORD}
    volumes:
      - pgadmin:/var/lib/pgadmin
    ports:
      - "5050:80"
    networks:
      - my_network

networks:
  my_network:
    driver: bridge

volumes:
  postgres-db-volume:
    external: false
  redis:
    external: false
  pgadmin:
    external: false